cookbook-react/recipes/inputs/input-3/index.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import cn from 'classnames';
+
+import './styles.scss';
+
+interface Props {
+  disabled?: boolean;
+  error?: string;
+}
+
+function Input({ disabled, error }: Props) {
+  return (
+    <div id="input-3" className="input-container">
+      <div className="input-wrapper">
+        <input className={cn("input-base", { "input-error": !!error, "input-disabled": disabled })} placeholder=" " disabled={disabled} />
+        <span className={cn("input-label", { "label-disabled": disabled })}>Label</span>
```

Comment:
Shouldn't you receive the label? :thinking:

Reply:
Btw, don't use spans as labels. If we have accessibility in mind, what we should do is to:
- Use labels for inputs.
- Link each label with the input by an id.
- Structurally speaking, labels should come before the inputs. If you wanna reorder them, you can do it by CSS.

It would be something like this:
```js
<label for='{inputId}' className={cn("input-label", { "label-disabled": disabled })}>{label}</label>
<input id='{inputId}' className={cn("input-base", { "input-error": !!error, "input-disabled": disabled })} placeholder=" " disabled={disabled} />
```

If you're interested in this topic, you can read more about it [in this site](https://www.w3.org/WAI/tutorials/forms/labels/).

Reply:
Keeping the same order because is required for the animation, for now.